src/julia-parser.scm

```diff
@@ -1271,7 +1271,11 @@
                 (syntax-deprecation s (string "abstract " (deparse spec))
                                     (string "abstract type " (deparse spec) " end")))
             (begin0 (list 'abstract spec)
-                    (if ty (expect-end s "abstract type"))))))
+                    (if ty (begin
+                      (let ((nt (peek-token s)))
+                        (if (or (newline? nt) (eqv? nt #\;))
+                            (take-token s)))
+                      (expect-end s "abstract type")))))))
```

Comment:
Good points, thanks! This is my first time doing anything nontrivial in Lisp so I haven't yet gotten the hang of how best to structure code. This is very helpful.